### PR TITLE
Initialise v2 modules

### PR DESCRIPTION
**Pull request by axw**
- Bump agent version to v2.0.0
- Bump all modules to v2.0.0, and add a v2 suffix to each
- Remove support for versions of Go older than 1.13 -- maybe we can go further, this is just a starting point.

See https://github.com/elastic/apm-agent-go/issues/654

Closes https://github.com/elastic/apm-agent-go/issues/1184

### CODE REVIEW

**Comment by axw** [comment]
This is blocked until go.elastic.co is updated, as otherwise we'll break links to the existing v1 modules.

**Review by marclop — APPROVED** [review]
LGTM after a couple of browser crashes 😅. I think there are just a couple of prefixes missing from the docs that I could see.

File: docs/instrumenting.asciidoc
Comment (axw) [reply docs/instrumenting.asciidoc:416]: indeed, thanks!

**Comment by simitt** [comment]
We could test the general upgrade experience by for example upgrading the apm agent in the opbeans service.

**Comment by axw** [comment]
opbeans-go has already been updated to use v2, and I think it went pretty smoothly :)

https://github.com/elastic/opbeans-go/pull/64/files

**Pull request merged**
